controllers/configurationpolicy_controller.go

```diff
@@ -1387,9 +1390,33 @@ func (r *ConfigurationPolicyReconciler) determineDesiredObjects(
 			return nil, &scopedGVR, errEvent, err
 		}
 
+		// Fetch object when:
+		// - Object var is used
+		// - Name is provided and not templated
+		// - Namespace is empty and not templated
+		// (Fetching for the object selector is handled later on)
+		fetchObject := needsObject && desiredName != "" && parsedMinMetadata.Metadata.Namespace == ""
+
 		if len(selectedNamespaces) != 0 {
 			for _, ns := range selectedNamespaces {
-				relevantNsNames[ns] = defaultNamesPerNs
+				switch {
```

Comment:
Is there a linter that was complaining about an `if` statement here? Using a `switch` seems odd for a boolean.